org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/preferences/BuilderPreferenceAccess.java

```diff
@@ -109,4 +140,22 @@ public void setAutoBuildEnabled(Object context, boolean enabled) {
 		preferenceStore.setValue(PREF_AUTO_BUILDING, enabled);
 	}
 
+	/**
+	 * @since 2.16
+	 */
+	public SchedulingOption getSchedulingRule (Object context) {
+		IPreferenceStore preferenceStore = preferenceStoreAccess.getContextPreferenceStore(context);
+		if (preferenceStore.contains(PREF_SCHEDULING_RULE)) {
+			return SchedulingOption.fromName(preferenceStore.getString(PREF_SCHEDULING_RULE));
+		}
+		return SchedulingOption.WORKSPACE;
+	}
+	
+	/**
+	 * @since 2.16
+	 */
+	public void setSchedulingRule (Object context, SchedulingOption opt) {
```

Comment:
what is this needed for?

Reply:
If you'd like to set this preference programmatically. I'd like to have that symetric. Option 'autoBuildEnabled' is also settable.

Reply:
see other place. this is a global and not a language specific option

Reply:
See above. From my understanding, this should be a workspace option and not per project.

Reply:
Agreed.